pkg/cmd/step/create/step_create_meta_pipeline.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	if o.PipelineKind == "" {
+		return util.MissingOption(pipelineKindOptionName)
+	}
+
+	if util.StringArrayIndex(jenkinsfile.PipelineKinds, o.PipelineKind) == -1 {
+		return util.InvalidOption(pipelineKindOptionName, o.PipelineKind, jenkinsfile.PipelineKinds)
+	}
+
+	_, err = util.ExtractKeyValuePairs(o.CustomLabels, "=")
+	if err != nil {
+		return errors.Wrap(err, "unable to parse custom labels")
+	}
+
+	_, err = util.ExtractKeyValuePairs(o.CustomEnvs, "=")
+	if err != nil {
+		return errors.Wrap(err, "unable to parse custom environment variables")
+	}
+
+	if o.PullRefs != "" {
+		_, err = prow.ParsePullRefs(o.PullRefs)
+		if err != nil {
+			return errors.Wrapf(err, "unable to parse pull ref '%s'", o.PullRefs)
+		}
+	}
+
+	return nil
+}
+
+func (o *StepCreatePipelineOptions) handleResult(tektonClient tektonclient.Interface,
+	jxClient jxclient.Interface,
+	tektonCRDs *tekton.CRDWrapper,
+	buildNumber string,
+	branch string,
+	pullRefs string,
+	ns string,
+	gitInfo *gits.GitRepository) error {
+
+	pr, err := o.buildPullRevs(pullRefs)
+	if err != nil {
+		return errors.Wrapf(err, "failed to build pull refs")
+	}
+	pipelineActivity := tekton.GeneratePipelineActivity(buildNumber, branch, gitInfo, pr)
+	if o.NoApply {
+		err := tektonCRDs.WriteToDisk(o.OutDir, pipelineActivity)
+		if err != nil {
+			return errors.Wrapf(err, "failed to output Tekton CRDs")
+		}
+	} else {
+		err := tekton.ApplyPipeline(jxClient, tektonClient, tektonCRDs, ns, gitInfo, o.Branch, pipelineActivity)
+		if err != nil {
+			return errors.Wrapf(err, "failed to apply Tekton CRDs")
+		}
+		if o.Verbose {
+			log.Logger().Infof("applied tekton CRDs for %s\n", tektonCRDs.PipelineRun().Name)
+		}
+	}
+	return nil
+}
+
+func (o *StepCreatePipelineOptions) getPodTemplates(kubeClient kubeclient.Interface, ns string, containerName string) (map[string]*corev1.Pod, error) {
+	podTemplates, err := kube.LoadPodTemplates(kubeClient, ns)
+	if err != nil {
+		return nil, err
+	}
+
+	return podTemplates, nil
+}
+
+func (o *StepCreatePipelineOptions) getClientsAndNamespace() (tektonclient.Interface, jxclient.Interface, kubeclient.Interface, string, error) {
+	tektonClient, _, err := o.TektonClient()
+	if err != nil {
+		return nil, nil, nil, "", errors.Wrap(err, "unable to create Tekton client")
+	}
+
+	jxClient, _, err := o.JXClient()
+	if err != nil {
+		return nil, nil, nil, "", errors.Wrap(err, "unable to create JX client")
+	}
+
+	kubeClient, ns, err := o.KubeClientAndDevNamespace()
+	if err != nil {
+		return nil, nil, nil, "", errors.Wrap(err, "unable to create Kube client")
+	}
+
+	return tektonClient, jxClient, kubeClient, ns, nil
+}
+
+func (o *StepCreatePipelineOptions) buildLabels(gitInfo *gits.GitRepository) map[string]string {
+	// start with custom labels
+	labels, _ := util.ExtractKeyValuePairs(o.CustomLabels, "=")
+
+	// add some labels we always want to have based on the build information available
+	labels["owner"] = gitInfo.Organisation
```

Comment:
Maybe we should propagate the `prowJobName` to both of the `PipelineRun`s at the very least? Or am I missing something and we already are?

Reply:
sure, if we need that.